app/src/androidTest/java/ch/epfl/skysync/viewmodel/FlightsViewModelTest.kt

```diff
@@ -0,0 +1,212 @@
+import ch.epfl.skysync.models.flight.Role
+import ch.epfl.skysync.models.flight.RoleType
+import ch.epfl.skysync.models.flight.Team
+import java.time.LocalDate
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class FlightsViewModelTest {
+
+  private val db = FirestoreDatabase(useEmulator = true)
+  private val dbSetup = DatabaseSetup2()
+  private val flightTable = FlightTable(db)
+  private val basketTable = BasketTable(db)
+  private val balloonTable = BalloonTable(db)
+  private val flightTypeTable = FlightTypeTable(db)
+  private val vehicleTable = VehicleTable(db)
+
+  @get:Rule val composeTestRule = createComposeRule()
+  lateinit var viewModel: FlightsViewModel
+
+  @Before
+  fun setUp() {
+    dbSetup.clearDatabase(db)
+    dbSetup.fillDatabase(db)
+    composeTestRule.setContent {
+      viewModel =
+          FlightsViewModel.createViewModel(
+              flightTable, balloonTable, basketTable, flightTypeTable, vehicleTable)
+
+      // val flights = viewModel.currentFlights.collectAsStateWithLifecycle()
+      // Text(flights.value.getOrNull(0)?.toString()?: "No flight types")
+    }
+  }
+
+  @Test
+  fun fetchesCurrentFlightsIfEmpty() {
+    val currentFlights = viewModel.currentFlights.value
+    Assert.assertEquals(currentFlights.size, 0)
+  }
+
+  @Test
+  fun fetchesCurrentFlights() {
+    var flight1 =
+        PlannedFlight(
+            nPassengers = 2,
+            team =
+                Team(
+                    roles =
+                        listOf(
+                            Role(RoleType.PILOT, dbSetup.pilot1),
+                            Role(RoleType.CREW, dbSetup.crew1))),
+            flightType = dbSetup.flightType2,
+            balloon = dbSetup.balloon1,
+            basket = dbSetup.basket2,
+            date = LocalDate.of(2024, 8, 12),
+            timeSlot = TimeSlot.AM,
+            vehicles = listOf(dbSetup.vehicle1),
+            id = UNSET_ID)
+
+    var flight2 =
+        PlannedFlight(
+            nPassengers = 2,
+            team =
+                Team(
+                    roles =
+                        listOf(
+                            Role(RoleType.PILOT, dbSetup.pilot1),
+                            Role(RoleType.CREW, dbSetup.crew1))),
+            flightType = dbSetup.flightType1,
+            balloon = dbSetup.balloon1,
+            basket = dbSetup.basket2,
+            date = LocalDate.of(2024, 8, 12),
+            timeSlot = TimeSlot.AM,
+            vehicles = listOf(dbSetup.vehicle1),
+            id = UNSET_ID)
+
+    flightTable.add(flight1, { flight1 = flight1.copy(it) }, { error -> println(error) })
+
+    flightTable.add(flight2, { flight2 = flight2.copy(it) }, { error -> println(error) })
+
+    SystemClock.sleep(DB_SLEEP_TIME)
+
+    viewModel.refreshCurrentFlights()
+    SystemClock.sleep(DB_SLEEP_TIME)
+    Assert.assertEquals(viewModel.currentFlights.value.size, 2)
```

Comment:
Assertions work the other way around. At the left there is the expected value, and on the right the actual value.